pastas/stressmodels.py

```diff
@@ -1151,3 +1153,155 @@ def to_dict(self, series=True):
+
+    See Also
+    --------
+    pastas.recharge
+
+    Notes
+    -----
+    The Threshold autoregressive self-exciting open-loop (Tarso) model is
+    nonlinear in structure because it incorporates two regimes which are
+    separated by a threshold. This model method can be used to simulate a
+    groundwater system where the groundwater head reaches the surface or
+    drainage level in wet conditions.
+    RechargeTarsoModel uses two drainage levels, with two exponential response
+    functions. When the simulation reaches the second drainage level, the
+    second response function becomes active. Because ofits structure,
+    RechargeTarsoModel cannot be combined with other stressmodels, a constant
+    or a transform.
+    RechargeTarsoModel inherits from RechargeModel. Only parameters specific to
+    the child class are named above. 
+
+    """
+    _name = "RechargeTarsoModel"
+    def __init__(self, prec, evap, oseries=None, dmin=None, dmax=None,
+                 rfunc=Exponential, **kwargs):
+        if oseries is not None:
+            if dmin is not None or dmax is not None:
+                msg = 'Please specify either oseries or dmin and dmax'
+                raise(Exception(msg))
+            o = TimeSeries(oseries).series
+            dmin = o.min()
+            dmax = o.max()
+        elif dmin is None or dmax is None:
+            msg = 'Please specify either oseries or dmin and dmax'
+            raise(Exception(msg))
+        if rfunc != Exponential:
+            raise(Exception('RechargeTarsoModel can only use an exponential '
+                            'response function'))
+        self.dmin = dmin
+        self.dmax = dmax
+        super().__init__(prec, evap, rfunc=rfunc, **kwargs)
+        
+    def set_init_parameters(self):      
+        # parameters for the first drainage level
+        p0 = self.rfunc.get_init_parameters(self.name)
+        one = One(meanstress = self.dmin + 0.5 * (self.dmax - self.dmin))
+        pd0 = one.get_init_parameters(self.name).squeeze()
+        p0.loc['{}_d'.format(self.name)] = pd0
+        p0.index = ['{}0'.format(x) for x in p0.index]
+        
+        # parameters for the second drainage level
+        p1 = self.rfunc.get_init_parameters(self.name)
+        initial = self.dmin + 0.75 * (self.dmax - self.dmin)
+        pd1 = Series({'initial':initial, 'pmin':self.dmin, 'pmax':self.dmax,
+                      'vary':True, 'name':self.name})
+        p1.loc['{}_d'.format(self.name)] = pd1
+        p1.index = ['{}1'.format(x) for x in p1.index]
+        
+        # parameters for the recharge-method
+        pr = self.recharge.get_init_parameters(self.name)
+        
+        # combine all parameters
+        self.parameters = concat([p0, p1, pr])
+    
+    def simulate(self, p=None, tmin=None, tmax=None, freq=None, dt=1):
+        stress = self.get_stress(p=p, tmin=tmin, tmax=tmax, freq=freq)
+        h = self.tarso(p[:-self.recharge.nparam], stress.values, dt)
```

Comment:
self.recharge.nparam??? I think this model should not inherit from RechargeModel. It will make it more difficult to maintain and doesn't add much.